functions/messages.js

```diff
@@ -179,6 +187,94 @@ _${safetyReminder}_
   return content;
 }
 
+const renderDeliveryDM = (ticketID, deliveryDMContent, deliveryChannel) => (
+  {
+    channel: deliveryChannel,
+    as_user: true,
+    text: deliveryDMContent, // fallback for blocks section
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: deliveryDMContent
+        }
+      },
+      {
+        type: 'actions',
+        elements: [
+          {
+            type: 'button',
+            text: {
+              type: 'plain_text',
+              text: 'Email me a shopping list'
+            },
+            style: 'primary',
+            action_id: 'email_shopping_list',
+            value: ticketID
+          }
+        ]
+      }
+    ],
+    unfurl_media: false,
+    unfurl_links: false,
+  }
+);
+
+async function getShoppingList(tickets) {
+  const itemsByHouseholdSize = await getItemsByHouseholdSize();
+  const categorizedStandardFoodOptions = ([, fields]) => {
+    return _.fromPairs(
+      _.map(fields.foodOptions, (item) => [
+        item,
+        {
+          category: (itemsByHouseholdSize[item]) ? itemsByHouseholdSize[item].category : 'Custom',
+          amounts: {
+            ticket: fields.ticketID,
+            quantity: (itemsByHouseholdSize[item]) ? itemsByHouseholdSize[item][fields.householdSize] : 'custom',
+          },
+          unit: (itemsByHouseholdSize[item]) ? itemsByHouseholdSize[item].unit : 'custom',
+        },
+      ])
+    );
+  };
+  const addAmounts = (item, acc, amounts) => {
+    return _.get(acc, item, { amounts: [] }).amounts.concat([amounts]);
+  };
+  const totalCategorizedStandardFoodOptions = _.reduce(
+    _.map(tickets, categorizedStandardFoodOptions),
+    (acc, groups) => {
+      const updates = _.fromPairs(
+        _.map(_.entries(groups), ([item, { category, amounts, unit }]) => [
+          item,
+          {
+            category,
+            amounts: addAmounts(item, acc, amounts),
+            unit,
+          },
+        ])
+      );
+      return Object.assign(acc, updates);
+    },
+    {}
+  );
+  const flattened = _.map(_.entries(totalCategorizedStandardFoodOptions), ([item, { category, amounts, unit }]) => ({ item, category, amounts, unit }));
+  return _.groupBy(flattened, 'category');
+}
+
+const renderShoppingList = (groups) => {
```

Comment:
On my branch (I think only on my laptop, I didn't push to github), I have a separate function that renders a shopping list as if you are only shopping for a single ticket (which I've been using myself, and like the format of better, because I don't shop for multiple households at once). I'd like to provide people with two buttons to click: "email me the shopping list and ticket metadata for just this one ticket", and "email me an aggregated shopping list for all my currently assigned tickets".

But, I think we can merge your PR first and then tweak it to do that.

Reply:
(yeah, I did push my changes to my branch, but after you created your branch from mine)

Reply:
okay cool if you're cool with doing that post merge I'm just going to test that this still works and go ahead